### PR TITLE
TIP-744: Set PHP 7.1 as minimum requirement

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

We can, finally, fully support only PHP 7.1 \o/

For that, PHP requirement in `composer.json` has been updated.

The `PimRequirements` class has been reworked too, to check MySQL version. We cannot use the old `mysql_get_client_info()` method as it does not exists anymore in PHP 7.x: https://secure.php.net/manual/fr/function.mysql-get-client-info.php
So I choose to use `PDO` (as we already check it is installed) and the PIM parameters to access MySQL and get its version.
The only other way to do it would have been to use mysqli extension, but it is not a PIM requirement, so it is not safe at all to rely on it. Some suggested to use a simple `exec(mysql --version)` but this cannot work on install where MySQL is on a different machine (distant MySQL server or docker installation with different containers).

I also updated the "storage data collector" (that did get MySQL version too), so it use PDO (and remove the MongoDB part of it).

Finally, there was some small fixes to do on behat.

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -104,15 +102,6 @@ if (launchIntegrationTests.equals("yes")) {
...
-        tasks["phpunit-7.0-akeneo"] = {runIntegrationTest("7.0", "Akeneo_Integration_Test")}
-        tasks["phpunit-7.0-api-base"] = {runIntegrationTest("7.0", "PIM_Api_Base_Integration_Test")}
-        tasks["phpunit-7.0-api-controllers"] = {runIntegrationTest("7.0", "PIM_Api_Bundle_Controllers_Integration_Test")}
-        tasks["phpunit-7.0-api-controllers-catalog"] = {runIntegrationTest("7.0", "PIM_Api_Bundle_Controllers_Catalog_Integration_Test")}
-        tasks["phpunit-7.0-api-controller-product"] = {runIntegrationTest("7.0", "PIM_Api_Bundle_Controller_Product_Integration_Test")}
-        tasks["phpunit-7.0-catalog"] = {runIntegrationTest("7.0", "PIM_Catalog_Integration_Test")}
-        tasks["phpunit-7.0-completeness"] = {runIntegrationTest("7.0", "PIM_Catalog_Completeness_Integration_Test")}
-        tasks["phpunit-7.0-pqb"] = {runIntegrationTest("7.0", "PIM_Catalog_PQB_Integration_Test")}
-
         tasks["phpunit-7.1-akeneo"] = {runIntegrationTest("7.1", "Akeneo_Integration_Test")}
```
Comment (jjanvier) [inline Jenkinsfile:105]: what about the other series?

**Review by jjanvier — COMMENTED** [review]
Is it normal to reject the user at this step if he has not prepared its database?

File: Jenkinsfile (hunk 1 above)
Comment (damien-carcel) [reply Jenkinsfile:105]: They are the same numbers, the diff just hides it :).

**Comment by damien-carcel** [comment]
Yes I think, as the file is created by `composer update`, and the requirement is usually ran during `pim:install`, so database has to be configured.

**Comment by damien-carcel** [comment]
Closed in favor of https://github.com/jjanvier/pim-community-dev/pull/242

**Pull request closed without merging**